File: 01.backend/src/main/java/com/fallapp/dto/VotoDTO.java
package com.fallapp.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * DTO para transferencia de datos de Voto
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VotoDTO {
    private Long idVoto;
    private Long idUsuario;
    private String nombreUsuario;
    private Long idFalla;
    private String nombreFalla;
    private String tipoVoto;
    private LocalDateTime fechaCreacion;
}

File: 03.mobile/app/src/main/java/com/fallapp/features/fallas/data/remote/dto/VotoDto.kt
package com.fallapp.features.fallas.data.remote.dto

import kotlinx.serialization.Serializable

/**
 * DTO para un voto desde el backend.
 */
@Serializable
data class VotoDto(
    val idVoto: Long,
    val idUsuario: Long,
    val nombreUsuario: String,
    val idFalla: Long,
    val nombreFalla: String,
    val tipoVoto: String,
    val fechaCreacion: String
)

/**
 * DTO para crear un voto.
 */
@Serializable
data class VotoRequestDto(
    val idFalla: Long,
    val tipoVoto: String
)

